### ISSUE TITLE
Reject `PROPFIND` requests with external entities with a `no-external-entities` precondition

### ISSUE THREAD

**Issue by jwodder** [dandi/dandidav#36]
[RFC 4918 requires](http://www.webdav.org/specs/rfc4918.html#n-implications-of-xml-entities) that when a `PROPFIND` request contains any XML external entities and the server rejects it as a result, the rejection should be a 403 response of the form:

```xml
<?xml version="1.0" encoding="utf-8"?>
<error xmlns="DAV:">
    <no-external-entities />
</error>
```

**Comment by yarikoptic** [comment]
did we encounter any of such, e.g. as triggered by rclone or some other FUSE system?

**Comment by jwodder** [comment]
@yarikoptic I don't believe so.  The only clients that would ever send bodies with external entities would be malicious.  At the moment, it seems that the xml-rs library that `dandidav` uses for parsing XML simply ignores any external entities.
